app/src/main/java/com/example/bookclub/QRCodeMainActivity.java

```diff
@@ -12,30 +13,48 @@
 import androidx.lifecycle.LifecycleOwner;
 
 import android.Manifest;
+import android.content.Intent;
 import android.content.pm.PackageManager;
 import android.os.Bundle;
+import android.util.Log;
+import android.util.Size;
+import android.view.View;
+import android.widget.Button;
 import android.widget.Toast;
 
 import com.bumptech.glide.util.MarkEnforcingInputStream;
+import com.example.bookclub.utils.QRCodeFoundListener;
 import com.google.common.util.concurrent.ListenableFuture;
 
 import java.util.concurrent.ExecutionException;
 
 public class QRCodeMainActivity extends AppCompatActivity {
+    public static final int PERMISSION_REQUEST_CAMERA = 0;
     private PreviewView mPreviewView;
     private ListenableFuture<ProcessCameraProvider> cameraProviderFuture;
-    public static final int PERMISSION_REQUEST_CAMERA = 0;
+    private Button mQRCodeFound;
+    private String qrCode;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_qrcode_main);
 
         mPreviewView = findViewById(R.id.qrcode_preview);
+        mQRCodeFound = findViewById(R.id.activity_main_qrcode_btn);
+        mQRCodeFound.setVisibility(View.INVISIBLE);
```

Comment:
You can make this GONE instead of invisible.

Reply:
Thanks!